src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -964,6 +971,22 @@ private Conversion GetSwitchExpressionConversion(BoundExpression source, TypeSym
             }
         }
 
+        private Conversion GetConditionalExpressionConversion(BoundExpression source, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            if (!(source is BoundUnconvertedConditionalOperator conditionalOperator))
```

Comment:
An excellent of example of why `not` should be the same and `!` :)

Reply:
It cannot be, except perhaps as a special case at the top level of the pattern immediately beneath an *is-pattern* operator. Even that would undermine the applicability of the demorgan's law transformation for patterns. Cyrus has queued that up for LDM discussion.

---
In reply to: [410419596](https://github.com/dotnet/roslyn/pull/40604#discussion_r410419596) [](ancestors = 410419596)